integration/container/nat_test.go

```diff
@@ -102,34 +91,22 @@ func startServerContainer(t *testing.T, msg string, port int) string {
 	client := request.NewAPIClient(t)
 	ctx := context.Background()
 
-	c, err := client.ContainerCreate(ctx,
-		&container.Config{
-			Image: "busybox",
-			Cmd:   []string{"sh", "-c", fmt.Sprintf("echo %q | nc -lp %d", msg, port)},
-			ExposedPorts: map[nat.Port]struct{}{
-				nat.Port(fmt.Sprintf("%d/tcp", port)): {},
-			},
-		},
-		&container.HostConfig{
-			PortBindings: nat.PortMap{
-				nat.Port(fmt.Sprintf("%d/tcp", port)): []nat.PortBinding{
-					{
-						HostPort: fmt.Sprintf("%d", port),
-					},
+	cID := container.Run(t, ctx, client, container.WithName("server"), container.WithCmd("sh", "-c", fmt.Sprintf("echo %q | nc -lp %d", msg, port)), func(c *container.TestContainerConfig) {
+		c.Config.ExposedPorts = map[nat.Port]struct{}{
```

Comment:
It would be more clear with functions here too

```go
container.Run(t, ctx, client, container.WithCmd(…),
  container.WithExposedPort(fmt.Sprintf("%d/tcp", port)),
  container.WithPortBinding(port),
  // container.WithPortBinding(anotherPort), // …
)
```

Reply:
Thanks @vdemeester. I added the `WithExposedPorts(port ...string)`. For PortBindings, the specification requires `map[Port][]PortBinding`, which is a map of port as the key, and a slice of `PortBinding` (`IP` + `Port`) as the value. I think it is quite lengthy to even put it in a helper function. On the other hand, most of the use cases are fairly simple (like the one used in the test). Maybe we could defer the implementation later if we find more cases?

Reply:
yeah probably 😉